snapcraft/internal/errors.py

```diff
@@ -166,6 +166,14 @@ class MissingGadgetError(SnapcraftError):
         'https://github.com/snapcore/snapd/wiki/Gadget-snap')
 
 
+class PluginOutdatedError(SnapcraftError):
+
+    fmt = 'This plugin is outdated: {message}'
+
+    def __init__(self, message):
```

Comment:
Yeah, but I have two problems with using the super class's `__init__` directly:

1. When there's only one or two options (both of which are required) it doesn't add meaning, just typing.
2. Typos in the kwargs doesn't result in even a linter error, but completely borks the exception. It's bitten me a few times.

I find subclassing and using positional (or kwargs the linter can pick up on) to be safer. If you disagree I can fix it.